### ISSUE TITLE
XWayland fails on Eoan

### ISSUE THREAD

**Issue by wmww** [MirServer/mir#1013]
On Ubuntu Eoan, current master fails to start the XWayland server. When Mir and xeyes are run with the following commands:
```
$ ./bin/mir_demo_server --x11-display-experimental 2
```
```
$ DISPLAY=:2 xeyes
```
Mir produces this result:
```
[2019-09-26 15:20:23.254950] <information> xwaylandserver: Starting Xwayland
[2019-09-26 15:20:23.255782] <information> xwaylandserver: set ok
[2019-09-26 15:20:23.255824] <information> xwaylandserver: set ok
(EE) 
Fatal server error:
(EE) listener function for opcode 3 of zxdg_output_v1 is NULL
(EE) 
[2019-09-26 15:20:23.265811] <information> xwaylandserver: Stalled start of Xserver, trying to start again!
[2019-09-26 15:20:23.265865] <information> xwaylandserver: Starting Xwayland
[2019-09-26 15:20:23.266784] <information> xwaylandserver: set ok
[2019-09-26 15:20:23.266827] <information> xwaylandserver: set ok
(EE) 
Fatal server error:
(EE) listener function for opcode 3 of zxdg_output_v1 is NULL
(EE) 
```
And then repeats this several several more times. When run in GDB it crashes. This issue does not appear to effect Bionic. I suspect it is related to a new version of XWayland or other package that uses a newer Wayland protocol version.

**Comment by RAOF** [comment]
This *looks* like an artefact of Mir sending an unrecognised event on the `zxdg_output_v1` interface; this is the same sort of crash we were seeing when we were sending v2 events on an interface bound as v1.

But that *shouldn't* be possible for `zxdg_output_v1`, because there's only one version?

**Comment by RAOF** [comment]
[Hello!](https://github.com/MirServer/mir/blob/391da119d1e0965f1b7c59b1a3cfbb64701332fa/src/server/frontend_wayland/xdg_output_v1.cpp#L139)

**Comment by RAOF** [comment]
https://github.com/MirServer/mir/blob/391da119d1e0965f1b7c59b1a3cfbb64701332fa/src/server/frontend_wayland/xdg_output_v1.cpp#L139

**Comment by RAOF** [comment]
Alternatively, depending on how Wayland counts its opcodes, this *may* be an XWayland bug, where it assumes that we're exposing `zxdg_output_v1` v3 and so won't send the `zxdg_output_v1.done` event.

**Comment by AlanGriffiths** [comment]
Fixed by #1026
